packages/taiko-client/prover/proof_producer/zkvm_producer.go

```diff
@@ -479,14 +479,18 @@ func (s *ZKvmProofProducer) requestBatchProof(
 	if len(output.ErrorMessage) > 0 || len(output.Error) > 0 {
 		return nil, fmt.Errorf("failed to get batch proof, msg: %s", output.ErrorMessage)
 	}
+	if output.Data == nil {
+		return nil, fmt.Errorf("unexpected structure error, response: %s", string(resBytes))
+	}
+
 	if output.Data.Status == ErrProofInProgress.Error() {
 		return nil, ErrProofInProgress
 	}
 	if output.Data.Status == StatusRegistered {
 		return nil, ErrRetry
 	}
 
-	if len(output.Data.Proof.Proof) == 0 {
+	if output.Data.Proof == nil || len(output.Data.Proof.Proof) == 0 {
```

Comment:
Set `Data         *RaikoProofDataV2 `json:"data,omitempty"`` then just check `output.Data.Proof == nil` is enough.

Reply:
Got it, but I still want to use this code.